app/components/checkOutForm.jsx

```diff
@@ -0,0 +1,154 @@
+          default:
+            setMessage("Something went wrong.");
+            break;
+        }
+      }
+
+    });
+  }, [stripe]);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!stripe || !elements) {
+      // Stripe.js hasn't yet loaded.
+      // Make sure to disable form submission until Stripe.js has loaded.
+      return;
+    }
+
+    setIsLoading(true);
+
+    const { error } = await stripe.confirmPayment({
+      elements,
+      confirmParams: {
+        // Make sure to change this to your payment completion page
+        return_url: "http://localhost:3000",
+      },
+    });
+
+    // This point will only be reached if there is an immediate error when
+    // confirming the payment. Otherwise, your customer will be redirected to
+    // your `return_url`. For some payment methods like iDEAL, your customer will
+    // be redirected to an intermediate site first to authorize the payment, then
+    // redirected to the `return_url`.
+    if (error.type === "card_error" || error.type === "validation_error") {
+      setMessage(error.message);
+    } else {
+      setMessage("An unexpected error occurred.");
+    }
+
+    setIsLoading(false);
+  };
+
+  const paymentElementOptions = {
+    layout: "tabs",
+  };
+
+  const router = useRouter();
+
+  const handleIncreaseAmount = (product) => {
+    const newCart = [...cart];
+    const ind = newCart.findIndex(item => item.asin === product.asin);
+    newCart[ind].quantity++;
+    setCart(newCart);
+  }
+  const handleDecreaseAmount = (product) => {
+    const newCart = [...cart];
+    const ind = newCart.findIndex(item => item.asin === product.asin);
+    newCart[ind].quantity--;
+    setCart(newCart);
+  }
+  const handleDeleteFromCart = (product) => {
+    const newCart = [...cart];
+    const clearCart = newCart.filter(item => item.asin !== product.asin);
+    setCart(clearCart);
+  }
+
+
+  const total = cart.reduce((acc, curr) => acc + curr.price * curr.quantity, 0);
+  const checkoutList = cart.map((product, ind) =>
+    <div key={ind} className='w-screen h-96 flex'>
+
+      <div className='w-1/3 h-full flex justify-center relative m-2'>
+        <Image src={product.imgUrl} alt='product image' fill={true} objectFit="cover" objectPosition="center" />
+      </div>
+      <div className='flex flex-col justify-center pl-4'>
+        <p className='ml-2 text-sm sm:text-2xl'>{product.productDescription}</p>
+        <div className='flex pl-2'>
+          <button onClick={() => handleDecreaseAmount(product)} className="text-sm sm:text-2xl border-2 border-black rounded-lg px-2">-</button>
+          <p className='m-2 text-sm sm:text-2xl'>Quantity: {product.quantity}</p>
+          <button onClick={() => handleIncreaseAmount(product)} className="text-sm sm:text-2xl border-2 border-black rounded-lg px-2">+</button>
+        </div>
+        <p className='m-2 text-sm sm:text-2xl'>Price: {product.quantity * product.price}$</p>
+      </div>
+    </div>
+
+  )
```

Comment:
is this ending parentheses part of the `checkoutList` function? on line 115 it looks like you are not starting the function logic with a `{`. That's a coding style preference but typically not one I am in favor of specifically for situations like these. I'd recommend adding the Rainbow brackets VSCode extension to your text editor to highlight these situations.

Reply:
okay, usually I am not adding those brackets at the beginning because then I would have to add return() explicitly, but sometimes it becomes a course of errors when you have to add the brackets but you forget to add the return() because you never do.